nimare/meta/cbma/base.py

```diff
@@ -0,0 +1,476 @@
+
+
+class CBMAEstimator(MetaEstimator):
+    """Base class for coordinate-based meta-analysis methods.
+
+    Parameters
+    ----------
+    kernel_transformer : :obj:`nimare.base.KernelTransformer`, optional
+        Kernel with which to convolve coordinates from dataset. Default is
+        ALEKernel.
+    *args
+        Optional arguments to the :obj:`nimare.base.MetaEstimator` __init__
+        (called automatically).
+    **kwargs
+        Optional keyword arguments to the :obj:`nimare.base.MetaEstimator`
+        __init__ (called automatically).
+    """
+
+    _required_inputs = {"coordinates": ("coordinates", None)}
+
+    def __init__(self, kernel_transformer, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Get kernel transformer
+        kernel_args = {
+            k.split("kernel__")[1]: v for k, v in kwargs.items() if k.startswith("kernel__")
+        }
+
+        # Allow both instances and classes for the kernel transformer input.
+        from ..kernel import KernelTransformer
+
+        if not issubclass(type(kernel_transformer), KernelTransformer) and not issubclass(
+            kernel_transformer, KernelTransformer
+        ):
+            raise ValueError(
+                'Argument "kernel_transformer" must be a kind of ' "KernelTransformer"
+            )
+        elif not inspect.isclass(kernel_transformer) and kernel_args:
+            LGR.warning(
+                'Argument "kernel_transformer" has already been '
+                "initialized, so kernel arguments will be ignored: "
+                "{}".format(", ".join(kernel_args.keys()))
+            )
+        elif inspect.isclass(kernel_transformer):
+            kernel_transformer = kernel_transformer(**kernel_args)
+        self.kernel_transformer = kernel_transformer
+
+    def _fit(self, dataset):
+        """
+        Perform coordinate-based meta-analysis on dataset.
+
+        Parameters
+        ----------
+        dataset : :obj:`nimare.dataset.Dataset`
+            Dataset to analyze.
+        """
+        self.dataset = dataset
+        self.masker = self.masker or dataset.masker
+        self.null_distributions_ = {}
+
+        ma_values = self.kernel_transformer.transform(
+            self.inputs_["coordinates"], masker=self.masker, return_type="array"
+        )
+
+        self.weight_vec_ = self._compute_weights(ma_values)
+
+        stat_values = self.compute_summarystat(ma_values)
+
+        # Determine null distributions for summary stat (OF) to p conversion
+        if self.null_method.startswith("analytic"):
+            self._compute_null_analytic(ma_values)
+        else:
+            self._compute_null_empirical(ma_values, n_iters=self.n_iters)
+        p_values, z_values = self._summarystat_to_p(stat_values, null_method=self.null_method)
+
+        images = {"stat": stat_values, "p": p_values, "z": z_values}
+        return images
+
+    def _compute_weights(self, ma_values):
+        return None
```

Comment:
Do we need this if it isn't an abstract method?

Reply:
Yes, because it serves as a dummy method for subclasses that don't take weights (e.g., `ALE`). It just allows us to avoid having different `_fit()` methods.